clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -277,12 +332,12 @@ public void testHeartbeatNotSentIfAnotherOneInFlight() {
 
         result = heartbeatRequestManager.poll(time.milliseconds());
```

Comment:
we need to sleep the interval before polling here again. If not, we're not getting HB because the interval hasn't expired (not because of in-flight as we want to assert on the following line)

Reply:
I see down below (ln 337-340) the sequence to validate that the interval expired and no HB is generated because of the inflight. I truly don't see the value in keeping this one then (ln 333-335), which is not related to in-flight at all (HB interval is tested in other funcs). Makes sense?